File: demo_app/app/src/main/java/com/bandyer/demo_android_sdk/adapter_items/UserSelectionItem.java
/*
 * Copyright (C) 2018 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.demo_android_sdk.adapter_items;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.CheckBox;

import com.bandyer.demo_android_sdk.R;
import com.mikepenz.fastadapter.items.AbstractItem;

import java.util.List;

import butterknife.BindView;

/**
 * A simple RecyclerView item used to display the user name with a checkbox as a cell in the list.
 */
public class UserSelectionItem extends AbstractItem<UserSelectionItem, UserSelectionItem.ViewHolder> {

    public final String name;

    public UserSelectionItem(String name) {
        this.name = name;
    }

    @Override
    public long getIdentifier() {
        return name.hashCode();
    }

    @Override
    public int getType() {
        return R.id.user_selection_item_id;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.user_selection_item;
    }

    @NonNull
    @Override
    public ViewHolder getViewHolder(@NonNull View v) {
        return new ViewHolder(v);
    }

    static class ViewHolder extends ButterKnifeViewHolder<UserSelectionItem> {

        @BindView(R.id.checkbox)
        CheckBox checkBox;

        ViewHolder(View view) {
            super(view);
        }

        @Override
        public void bindView(final @NonNull UserSelectionItem item, @NonNull List<Object> payloads) {
            checkBox.setChecked(item.isSelected());
            checkBox.setText(item.name);
        }

        @Override
        public void unbindView(@NonNull UserSelectionItem item) {
            checkBox.setChecked(false);
            checkBox.setText(null);
        }
    }
}

File: app_utilities/src/main/java/com/bandyer/app_utilities/adapter_items/UserSelectionItem.kt
/*
 * Copyright (C) 2020 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */
package com.bandyer.app_utilities.adapter_items

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bandyer.app_utilities.R
import com.mikepenz.fastadapter.items.AbstractItem
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.user_selection_item_layout.view.*

/**
 * A simple RecyclerView item used to display the user name with a checkbox as a cell in the list.
 */
class UserSelectionItem(@JvmField val name: String) : AbstractItem<UserSelectionItem.ViewHolder>() {

    override var identifier: Long = name.hashCode().toLong()
    override val type: Int = R.id.user_selection_item_id
    override val layoutRes: Int= R.layout.user_selection_item_layout
    override fun getViewHolder(v: View): ViewHolder = ViewHolder(v)

    override fun bindView(holder: ViewHolder, payloads: List<Any>) {
        super.bindView(holder, payloads)
        holder.containerView.checkbox!!.isChecked = isSelected
        holder.containerView.checkbox!!.text = name
    }

    override fun unbindView(holder: ViewHolder) {
        super.unbindView(holder)
        holder.containerView.checkbox!!.isChecked = false
        holder.containerView.checkbox.text = null
    }

    class ViewHolder(val containerView: View) : RecyclerView.ViewHolder(containerView)

}